keras_cv/training/simclr/simclr_trainer.py

```diff
@@ -0,0 +1,189 @@
+            Note that this should be specified iff training with labeled images.
+            If provided, `classes` must be provided.
+        classes: optional number of classes to classify images into, only to be
+            specified if `include_top` is True.
+        projection_width: an optional width for the projection MLP in the SimCLR architecture
+        augmenter: a preprocessing layer to randomly augment input images for contrastive learning.
+            If no custom augmenter is provided, a default augmenter will be used,
+            and `value_range` must be provided.
+        value_range: the range of values the incoming images will have.
+            Required iff no `augmenter` is provided.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.
+
+    Returns:
+      A `keras.Model` instance.
+
+    """
+
+    def __init__(
+        self,
+        encoder,
+        include_probe,
+        classes=None,
+        projection_width=128,
+        augmenter=None,
+        value_range=None,
+    ):
+        super().__init__()
+
+        self.include_probe = include_probe
+        self.projection_width = projection_width
+
+        if not augmenter and not value_range:
+            raise ValueError("`value` is required when using the default augmenter.")
+        self.augmenter = augmenter or preprocessing.Augmenter(
+            [
+                preprocessing.RandomFlip("horizontal"),
+                preprocessing.RandomTranslation(0.25, 0.25),
+                preprocessing.RandomZoom((-0.5, 0.0), (-0.5, 0.0)),
+                preprocessing.RandomColorJitter(
+                    value_range=value_range,
+                    brightness_factor=0.5,
+                    contrast_factor=0.5,
+                    saturation_factor=(0.3, 0.7),
+                    hue_factor=0.5,
+                ),
+            ]
+        )
+
+        self.encoder = encoder
+
+        self.projection_top = keras.Sequential(
+            [
+                layers.Dense(self.projection_width, activation="relu"),
+                layers.Dense(self.projection_width),
+            ],
+            name="projection_top",
+        )
+
+        if self.include_probe:
+            if not classes:
+                raise ValueError(
+                    "`classes` must be specified when `include_probe` is `True`."
+                )
+            self.probing_top = layers.Dense(classes, name="linear_probe")
+
+    def compile(self, optimizer, loss, probe_optimizer=None, **kwargs):
+        super().compile(**kwargs)
+
+        self.optimizer = optimizer
+        self.loss = loss
+        self.simclr_loss_metric = keras.metrics.Mean(name="simclr_loss")
+
+        if self.include_probe:
+            self.probe_loss = keras.losses.SparseCategoricalCrossentropy(
```

Comment:
Let's add `probe_loss` as a compilation argument and make it required if include_probe=True